seacrowd/sea_datasets/culturax/culturax.py

```diff
@@ -0,0 +1,146 @@
+from seacrowd.utils.constants import Tasks
+
+_CITATION = """\
+@article{nguyen2023culturax,
+    author    = {Thuat Nguyen and Chien Van Nguyen and Viet Dac Lai and Hieu Man and Nghia Trung Ngo and Franck Dernoncourt and Ryan A. Rossi and Thien Huu Nguyen},
+    title     = {CulturaX: A Cleaned, Enormous, and Multilingual Dataset for Large Language Models in 167 Languages},
+    journal   = {arXiv preprint arXiv:2309.09400},
+    year      = {2023},
+    url       = {https://arxiv.org/abs/2309.09400},
+}
+"""
+
+_DATASETNAME = "culturax"
+_DESCRIPTION = """\
+CulturaX is a comprehensive multilingual dataset comprising 6.3 trillion tokens across 167
+languages, designed for large language model development. It incorporates an advanced
+cleaning and deduplication process, including language identification and fuzzy
+deduplication with MinHash, to ensure high-quality data for model training. The dataset,
+which spans 16TB in parquet format and 27TB when unpacked, is a combination of the latest
+mC4 and OSCAR corpora, emphasizing non-English languages to support multilingual model
+training. For data cleaning validation, CulturaX employs a SentencePiece tokenizer and
+KenLM language models, utilizing recent Wikipedia dumps for perplexity scoring.
+"""
+
+_HOMEPAGE = "https://huggingface.co/datasets/uonlp/CulturaX"
+_LICENSE = """
+    The licence terms for CulturaX strictly follows those of mC4 and OSCAR. \
+    Please refer to both below licenses when using this dataset. \
+    - mC4 license: https://huggingface.co/datasets/allenai/c4#license \
+    - OSCAR license: https://huggingface.co/datasets/oscar-corpus/OSCAR-2301#licensing-information \
+"""
+_BASE_URL = "https://huggingface.co/datasets/uonlp/CulturaX/resolve/main/{lang}/"
+
+_SUPPORTED_TASKS = [Tasks.SELF_SUPERVISED_PRETRAINING]
+_SOURCE_VERSION = "1.0.0"
+_SEACROWD_VERSION = "1.0.0"
+
+
+class CulturaXDataset(datasets.GeneratorBasedBuilder):
+    """CulturaX subset for SEA languages."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    SEACROWD_SCHEMA_NAME = "ssp"
+    SUBSETS = ["id", "jv", "km", "lo", "tl", "min", "my", "su", "th", "vi", "ms", "ceb", "war", "cbk", "bcl"]
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_{subset}_source",
+            version=datasets.Version(_SOURCE_VERSION),
+            description=f"{_DATASETNAME} {subset} source schema",
+            schema="source",
+            subset_id=subset,
+        ) for subset in SUBSETS
+    ] + [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_{subset}_seacrowd_ssp",
+            version=datasets.Version(_SEACROWD_VERSION),
+            description=f"{_DATASETNAME} {subset} SEACrowd schema",
+            schema="seacrowd_ssp",
+            subset_id=subset,
+        )
+        for subset in SUBSETS
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_jv_source"
+
```

Comment:
In the [tatoeba PR](https://github.com/SEACrowd/seacrowd-datahub/pull/22), the default config is a combination of all languages loaded into one. I'm not sure if we should do the same thing here (thoughts @SamuelCahyawijaya ?). I'm ok loading an arbitrary subset because CulturaX is quite big.

Reply:
Do all languages here mean all the approved SEA languages, or the whole dataset? Still though yeah, it's really big.

Reply:
Hello, this is a similar issue as the one mentioned [here](https://github.com/SEACrowd/seacrowd-datahub/pull/57), we agree not to enforce the default config that loads all datasets, i.e., `<DATASET>_source` and `<DATASET>_seacrowd_*`. In this case, we can just keep the current one (`f"{_DATASETNAME}_jv_source"`) or any other subset as the default config.